x-pack/plugins/security_solution/server/endpoint/services/actions/clients/sentinelone/sentinel_one_actions_client.ts

```diff
@@ -243,6 +255,8 @@ export class SentinelOneActionsClient extends ResponseActionsClientImpl {
       throw new ResponseActionsClientError(`SentinelOne agent id [${agentUUID}] not found`, 404);
     }
 
+    this.cache.set(agentUUID, s1ApiResponse.data[0]);
```

Comment:
Thanks @pzl for the feedback.

> re: `ttl`

the `ttl` on `.set()` method is optional. If undefined, the default `ttl` is used (see class constructor options).

</br></br>


> re: `cleanup()`

You are correct that `cleanup()` is not called in this usage of the `SImpleMemCache` - for a few reason (TL;DR; - its not really necessary):

- The class instance of SentinelOne is short lived... only used for a single request, so we're talking a few milliseconds/seconds
- The cached entries are auto-cleaned up when one attempts to `.get()` them... see implementation of the `.get() `method 


FYI: I added the `cleanup()` method so that if this class is ever used outside of the current use case (ex. in a long-lived class instance somewhere), then the `cleanup()` method can be used in a background timer (`setTimeout()`) to periodically go through the cache and clean up expired entries. 

Hope this makes sense. Let me know if you have any other questions or issue that is preventing you from 👍  it

Reply:
if the instance is only around for a request lifecycle, then yeah, none of these are concerns. and I totally missed the default value on ttl. But if its destroyed at the end of the request, everythings a-ok.

Did we even need a cache for something with a per-request lifespan?

Reply:
> Re: Did we even need a cache for something with a per-request lifespan?

There are multiple methods that need to access the agent details data from SentinelOne... that data requires an API request to the external system (via the Connector). I could have made the S1 class more complex by changing the method signatures so that I could then pass the agent details around between them, but I found that to be the least maintainable solution... So instead, I add the cache mechanism to it with a short `ttl` (default is `10s`) so that if the request does take a bit of time (maybe API calls to external system are slow), then the cache entry will expire and we'll make the API call again.